generate.py

```diff
@@ -38,16 +40,16 @@ def writefile(filename, s):
     bn = basename(sys.argv[0])
     s = '// DO NOT EDIT - autogenerated by ' + bn + '\n\n' + s
     if os.path.exists(filename) and open(filename).read() == s:
-        print 'Checked', filename
+        print('Checked', filename)
     else:
-        print 'Writing', filename
+        print('Writing', filename)
         open(filename, 'w').write(s)
 
 
 def generate_enum(enum_name, enum_prefix, count_name, name_list, value_list):
     # Can only generate an enum which starts at 0, and where each value is 1 more than the former
     assert len(name_list) == len(value_list)
-    assert value_list == list(xrange(0, len(value_list)))
+    assert value_list == list(range(0, len(value_list)))
```

Comment:
[Using `builtins` requires that you install `future`, too.](http://python-future.org/reference.html?highlight=builtins#module-future.types)

But as stated, there's generally only a small overhead on Python 2.